### DISCUSSION TITLE
1) Power Spikes on Zwift and 2) loss of SS2K network connection

### DISCUSSION THREAD

**Discussion by Inotrope88** [doudar/SmartSpin2k#399] [General]
Hi,

Been using the SS2k for a few months; recently getting some problems with the SS2k and Zwift:

Using FW 22.10.25 (rolled back from 22.10.27 as had same problems).
Assioma DUO pedals (latest FW) and Polar HR10 via BT to SS2K; Zwift on iPad. 

1) Power spikes - 20-30w spikes occurring every few seconds during races (picture https://www.dropbox.com/s/ev75xs4g2zp3jrh/IMG_1265.JPG?dl=0 )

2) I’m no longer able to connect to the SS2K via the browser, only via the direct connection to the SS2K on its WAP; any settings on the SS2k are lost so when I connect again via the WAP the SS2k displays default settings.

This is a bit odd, as all worked perfectly until recently so I wonder if this behaviour is a firmware problem?

I tried rolling back to 22.9.15 using the flashing software, but the SS2k updated itself to 22.10.25.

I use the SS2K a couple of times a week and don’t leave the powered on and connected.
 
Any thoughts appreciated, the power spikes are making my FTP higher than I think it should be!

**Comment by eMadman** [comment]
>     1. Power spikes - 20-30w spikes occurring every few seconds during races (picture https://www.dropbox.com/s/ev75xs4g2zp3jrh/IMG_1265.JPG?dl=0 )

Out of curiosity, do you have a second bluetooth power source in the room?  Like, does your bike transmit over BT?  It is possible that you're connected to both.  I've seen interesting spikes like that when my Bowflex C6 and tempo power meter were both connected at the same time.  In the bluetooth selection screen, make sure to select your assiomas if "Any" is selected.

Other than that, the other thing I can think of is... does this happen with your assiomas when they're connected directly to Zwift?
 
>     2. I’m no longer able to connect to the SS2K via the browser, only via the direct connection to the SS2K on its WAP; any settings on the SS2k are lost so when I connect again via the WAP the SS2k displays default settings.
Do you use the same SSID for both 2.4ghz and 5ghz on your wifi AP?  Can you try a separate SSID for 2.4ghz?  I had to do that due to a couple of my WIFI lights, relays, and power outlets that refused to play nicely any other way with my home network.

**Comment by Inotrope88** [comment]
ThanksYes the IC8 does transmit over BT - i do always make sure that the Assioma are selected I. The drop-down, and the drop down is saved.I’ll try racing with the Assioma connected directly to the iPad, and the SS2K doing the controller things and report back. It’s a different SSID for both the 2.4 and 5 gHz networks. Oddly this is all new behaviour on the SS2K and no kit or router changes has happened - other than firmware updates on the SS. . MarkSent from my iPhone XIV--------------------------------------------------------------Dr Mark Barley BMedSci BM BS MRCP FRCAConsultant Anaesthetist & Airway LeadNottingham University HospitalsDepartmental PA: 0115 9249924 ext: 61195--------------------------------------------------------------On 11 Nov 2022, at 01:55, eMadman ***@***.***> wrote:﻿

1. Power spikes - 20-30w spikes occurring every few seconds during races (picture https://www.dropbox.com/s/ev75xs4g2zp3jrh/IMG_1265.JPG?dl=0 )


Out of curiosity, do you have a second bluetooth power source in the room?  Like, does your bike transmit over BT?  It is possible that you're connected to both.  I've seen interesting spikes like that when my Bowflex C6 and tempo power meter were both connected at the same time.  In the bluetooth selection screen, make sure to select your assiomas if "Any" is selected.
Other than that, the other thing I can think of is... does this happen with your assiomas when they're connected directly to Zwift?

2. I’m no longer able to connect to the SS2K via the browser, only via the direct connection to the SS2K on its WAP; any settings on the SS2k are lost so when I connect again via the WAP the SS2k displays default settings.


Do you use the same SSID for both 2.4ghz and 5ghz on your wifi AP?  Can you try a separate SSID for 2.4ghz?  I had to do that due to a couple of my WIFI lights, relays, and power outlets that refused to play nicely any other way with my home network.

**Reply by eMadman** [reply]
Okay, cool - I think the spikiness you're seeing is a connection to both devices then.  I experienced something nearly identical when I first got my Tempo power meter.  You'll be able to validate by riding Zwift with just your ipad and assiomas.  Next test should be with the assiomas in the dropdown.  

Regarding your wifi issue, since it's worked in the past, I'm curious if the setting is corrupted on the microcontroller.  Would you be willing to try resetting to defaults?  This is on the Settings screen.  Just be sure to take a screenshot of your settings so you can get everything back the way you like it.

As a heads up, when you downgrade the firmware, you need to deselect the "automatic updates" option in the settings screen since it will try to install the latest firmware with each boot of the device.

**Comment by Inotrope88** [comment]
Thanks for the reply; and trouble shooting ideas!

The “spikiness” (perhaps oddly) seemed to have been related to very low power from my Polar HR10 HRM which I connect to the SS2k along with the Assioma; I’ve had some problems with the HR10 when the battery is nearly exhasted and it produces crazy heart rate outputs before - but as to why it would affect the power output data I don’t know!

I managed to fix the wifi problem by rebooting the local wireless access point the SS2k connects too.


I did have a problem with a loss of connection during ERG mode today which happened suddenly no idea why; is there a link to the number of devices one can connect to an iPad? 

Would it be better to connect the ss2k to a Mac for instance?
Mark

Sent from iCloud
------------------------------------------------------------
------------------------------------------------------------

**Reply by eMadman** [reply]
The ipad might have a maximum connection limit, but it's hard to say.  The SS2K is a multiplexer so if you have your HRM and bike connected to it, the ipad will only use one active connection - it passes everything along.  

If the connection loss happens mid-ride, you can quickly recover by pressing and holding both shifter buttons at the same time for 3 seconds.  This triggers a bluetooth reconnect.

If it's a regular issue, the laptop might be worth trying.

**Comment by Inotrope88** [comment]
Will try a laptop for tonights Zwift race!

Interestingly; I’d tried the 3 second bluetooth reconnect, but although the HR and Power were reported, the SS2K didn’t alter resistance at all which was puzzling (and a bit of a problem during an ERG workout!).
Any thoughts as to why that might have happened?

**Reply by doudar** [reply]
> Will try a laptop for tonights Zwift race! Interestingly; I’d tried the 3 second bluetooth reconnect, but although the HR and Power were reported, the SS2K didn’t alter resistance at all which was puzzling (and a bit of a problem during an ERG workout!). Any thoughts as to why that might have happened?
> […](#)
> On 15 Nov 2022, at 02:01, eMadman ***@***.***> wrote: The ipad might have a maximum connection limit, but it's hard to say. The SS2K is a multiplexer so if you have your HRM and bike connected to it, the ipad will only use one active connection - it passes everything along. If the connection loss happens mid-ride, you can quickly recover by pressing and holding both shifter buttons at the same time for 3 seconds. This triggers a bluetooth reconnect. If it's a regular issue, the laptop might be worth trying. — Reply to this email directly, view it on GitHub <[#399 (reply in thread)](https://github.com/doudar/SmartSpin2k/discussions/399#discussioncomment-4142203)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AYNRPV7ZV7UEZBRUGEDRRZ3WILVGTANCNFSM6AAAAAAR5A6BXM>. You are receiving this because you authored the thread.

We have a min/max position routine that attempts to estimate the minimum and maximum positions of the spin bike knob by continously collecting power and stepper position data and then using that to interpolate where the min_brake_watts and Max_brake_watts user settings are in relation to the stepper position. If the commanded position lies outside the bounds set by min/max, the stepper is halted in bounds. As more data is collected while you pedal, this interpolation gets progressively more accurate. It's not a perfect function (yet?) but it is better than the alternative of not having any end stops. Without these (software) end stops, we crash into the physical restraints of how far the knob on your bike can turn and subsequently lose positional accuracy due to the lost steps from the motor skipping. 

You can help to avoid this problem by pedaling in a ramping warmup before an ERG workout. Once the SS2K collects information on knob positions from 50w up to 150w, it should have enough information to calculate the end stops better. If you experience this in sim mode, you can unintuitivly shift in the opposite direction to get it to (hopefully) recalculate faster. 

Another method is to set both min and max brake_watts to 0 which should disable this function. 

I'm continually improving this function, with some updates to it incorporated into the last develop release binary which was built last night. You can manually install it if you'd like via the web interface. See the wiki for details on how to do that.

**Comment by Inotrope88** [comment]
Thanks, that’s interesting and sound really innovative.


An issue I’m getting repeatedly is that the power output to Zwift via the SS2K is greater than it should be!
(For instance, after a group cycle today, Zwift decided to increase my FTP to 329 - giving me 4.5w/kg! - my actual FTP is 286 )

I am using Assioma Duo pedals, configured to pass all data via the left pedal (the default).
I connect to Zwift via the SS2K along with my HR10
All devices connected ok.

The power output adjustment is set to 1x on the SS2K.


- Is there any way that the SS2K is somehow artificially increasing the power output as it acts as passthrough to Zwift?

- Should I connect the Assioma directly to Zwift and still use the SS2K as a controller (are there disadvantages to not using the SS2K as a passthrough for power; does the power/resistance interpolation function described below still function?)

- Should I always reconfigure the SS2K before every session? I’ve noticed that I cannot always reconnect to it via the browser if I’ve not used the SS2K for some hours, and I need to fully reboot the SS2k, then reconfigure the SS2K via its WAP beforeI can connect via a browser (all settings seem to be lost at this point)

**Reply by doudar** [reply]
The power isn't (shouldn't be) adjusted in SS2K by anything other than the power correction factor. I believe the auto-ftp in Zwift is a known issue in Zwift. 

You can directly connect the pedals to Zwift for power information. Sim mode will continue to work but ERG will not function. 

You only need access to WiFi whenever you need to make settings changes.

**Reply by MarkusSchneider** [reply]
The Assioma pedals supports multiple connections at the same time (at least two one via BLE and one by ANT+).
In my setup I'm using an ANT+ Dongle to connect the pedals to the pc or bike computer and BLE for SS2K.

If you connect the pedals directly to zwift via ANT+ Dongle, the values are transmitted a little bit faster.

**Reply by doudar** [reply]
> The Assioma pedals supports multiple connections at the same time (at least two one via BLE and one by ANT+).
> 
> In my setup I'm using an ANT+ Dongle to connect the pedals to the pc or bike computer and BLE for SS2K.
> 
> 
> 
> If you connect the pedals directly to zwift via ANT+ Dongle, the values are transmitted a little bit faster.
> 
> 

@MarkusSchneider , do the values you see on the SS2K differ from what you're seeing on Zwift?

**Reply by MarkusSchneider** [reply]
No values matches. I noticed only a slight delay.
